site/content/contribute/server/dependencies.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "Dependencies"
+date: 2019-03-27T16:00:00-0700
+weight: 5
+subsection: Server
+---
+
+
+The Mattermost server uses [go modules](https://github.com/golang/go/wiki/Modules) to manage dependencies. To manage dependencies you must have modules enabled.
+
+
+## Enabling go modules support (Do this first!)
+
+While golang migrates to support go modules, if you are developing in the GOPATH you need to set the environment variable `GO111MODULE` to `on` for example:
+
+```bash
+export GO111MODULE=on
+```
+
+Once this is done, all go commands issued will use the modules system.
+
+## Adding a new dependency
+
+Be sure you have enabled go modules support. Adding a dependency is easy. All you have to do is import the dependency in the code like normal and recompile. The dependency will be automatically added for you.
+
+If you want to add a specific version of a dependency you can use a command of the form:
+```bash
+go get -u <mydependency>@v0.0.0
+go mod tidy
+go mod vendor
+```
+For example:
+```bash
+go get -u github.com/mattermost/mattermost-server@v5.10.0
```

Comment:
Suggest illustrating with something like https://github.com/pkg/errors -- otherwise it's a confusing "example" of how to version yourself as a dependency (at least when updating server dependencies).